packages/wallet/src/types/types.ts

```diff
@@ -61,3 +67,57 @@ export interface SignatureProvider {
 export interface State {
   [address: string]: Storage
 }
+
+export type InclusionProof = string[]
+
+export interface StateInclusionProof {
+  [address: string]: InclusionProof
+}
+
+export interface StateUpdate {
+  transactions: SignedTransaction[]
+  startRoot: string
+  endRoot: string
+  updatedState: State
+  updatedStateInclusionProof: StateInclusionProof
+}
+
+export interface RollupTransition {
+  number: number
+  blockNumber: number
+  transactions: SignedTransaction[]
+  startRoot: string
+  endRoot: string
+}
+
+export interface RollupBlock {
+  number: number
+  transitions: RollupTransition[]
+}
+
+export interface TransactionReceipt {
+  blockNumber: number
+  transitionIndex: number
+  transaction: SignedTransaction
+  startRoot: string
+  endRoot: string
+  updatedState: State
+  updatedStateInclusionProof: StateInclusionProof
+}
+
+export interface SignedTransactionReceipt {
+  transactionReceipt: TransactionReceipt
+  signature: Signature
+}
+
+export interface StateReceipt {
+  address: string
```

Comment:
Just to confirm, will this `address` always be the same as the key to the `state` below, right?  Is it just there to make looking up easier?

On a separate note--I think we'll want a transition/block number here being signed too, so that the aggregator attestations are easier to penalize for being faulty--without them, it's a little trickier to handle.

Reply:
Yep easy lookup and as a way to say "I'm guaranteeing _this_ state is as follows".

Reply:
Also, added block number and transition index to StateReceipt 👍